docs/data/material/migration/migrating-to-v6/migrating-to-v6.md

```diff
@@ -1,17 +1,37 @@
-# Migrating to v6
+# Upgrade to v6
 
-<p class="description">This guide explains how and why to migrate from Material UI v5 to v6.</p>
+<p class="description">This guide explains why and how to upgrade from Material UI v5 to v6.</p>
+
+## Why you should upgrade to Material UI v6
+
+### React Server Component support
+
+Material UI v6 introduces Pigment CSS, a zero-runtime CSS-in-JS styling engine to replace Emotion and styled-components as a more future-proof solution for writing styles in React 19 and beyond.
+With Pigment CSS, styles are extracted at build time rather than runtime, avoiding client-side recalculations and unlocking React Server Component (RSC) compatibility.
+This also leads to significant reductions in bundle sizes for Material UI apps.
+
+**In v6, Pigment CSS migration is opt-in.**
+Future major versions of Material UI will likely use Pigment CSS as the default styling solution.
+Though optional, we highly recommend migrating your Material UI app to Pigment CSS once you've upgraded to v6.
```

Comment:
After chatting with @siriwatkn, @brijeshb42, and the rest of the team, we think we should tweak the wording a bit. Given that Pigment CSS is in alpha and areas like performance are under active development, the idea is to push consumers to try it out but avoid things like "highly recommended" until Pigment is ready for prime time.

We should take this into account in other docs pages, blog posts, etc.

Reply:
Changed "highly recommend" to "encourage"